tests/compute/test_new_kvstore.py

```diff
@@ -0,0 +1,161 @@
+                       init_func=init_zero_func)
+    # Test data_name_list
+    name_list = kvclient.data_name_list()
+    assert 'data_0' in name_list
+    assert 'data_1' in name_list
+    assert 'data_2' in name_list
+    # Test get_meta_data
+    meta = kvclient.get_data_meta('data_0')
+    dtype, shape, policy = meta
+    assert dtype == F.dtype(data_0)
+    assert shape == F.shape(data_0)
+    assert policy.policy_str == 'node'
+    meta = kvclient.get_data_meta('data_1')
+    dtype, shape, policy = meta
+    assert dtype == F.dtype(data_1)
+    assert shape == F.shape(data_1)
+    assert policy.policy_str == 'edge'
+    meta = kvclient.get_data_meta('data_2')
+    dtype, shape, policy = meta
+    assert dtype == F.dtype(data_2)
+    assert shape == F.shape(data_2)
+    assert policy.policy_str == 'node'
+    # Test push and pull
+    id_tensor = F.tensor([0,2,4], F.int64)
+    data_tensor = F.tensor([[6.,6.],[6.,6.],[6.,6.]], F.float32)
+    kvclient.push(name='data_0',
+                  id_tensor=id_tensor,
+                  data_tensor=data_tensor)
+    kvclient.push(name='data_1',
+                  id_tensor=id_tensor,
+                  data_tensor=data_tensor)
+    kvclient.push(name='data_2',
+                  id_tensor=id_tensor,
+                  data_tensor=data_tensor)
+    res = kvclient.pull(name='data_0', id_tensor=id_tensor)
+    assert_array_equal(F.asnumpy(res), F.asnumpy(data_tensor))
+    res = kvclient.pull(name='data_1', id_tensor=id_tensor)
+    assert_array_equal(F.asnumpy(res), F.asnumpy(data_tensor))
+    res = kvclient.pull(name='data_2', id_tensor=id_tensor)
+    assert_array_equal(F.asnumpy(res), F.asnumpy(data_tensor))
+    # Register new push handler
+    kvclient.register_push_handler(udf_push)
+    # Test push and pull
+    kvclient.push(name='data_0',
+                  id_tensor=id_tensor,
+                  data_tensor=data_tensor)
+    kvclient.push(name='data_1',
+                  id_tensor=id_tensor,
+                  data_tensor=data_tensor)
+    kvclient.push(name='data_2',
+                  id_tensor=id_tensor,
+                  data_tensor=data_tensor)
+    data_tensor = data_tensor * data_tensor
+    res = kvclient.pull(name='data_0', id_tensor=id_tensor)
+    assert_array_equal(F.asnumpy(res), F.asnumpy(data_tensor))
+    res = kvclient.pull(name='data_1', id_tensor=id_tensor)
+    assert_array_equal(F.asnumpy(res), F.asnumpy(data_tensor))
+    res = kvclient.pull(name='data_2', id_tensor=id_tensor)
+    assert_array_equal(F.asnumpy(res), F.asnumpy(data_tensor))
+    # clean up
+    dgl.distributed.shutdown_servers()
+    dgl.distributed.finalize_client()
+
+@unittest.skipIf(os.name == 'nt', reason='Do not support windows yet')
+def test_kv_store():
+    ip_config = open("ip_config.txt", "w")
+    ip_config.write('127.0.0.1 30050 1\n')
+    ip_config.close()
+    pid = os.fork()
+    if pid == 0:
+        start_server()
+    else:
+        time.sleep(1)
+        start_client()
```

Comment:
can you test backup servers? i think this can still be tested in a single machine.

Reply:
Sure.